lib/cli-engine.js

```diff
@@ -475,8 +475,8 @@ CLIEngine.prototype = {
      */
     resolveFileGlobPatterns: function(patterns) {
         var extensions = this.options.extensions.map(function(ext) {
-                return ext.charAt(0) === "." ? ext.substr(1) : ext;
-            });
+            return ext.charAt(0) === "." ? ext.substr(1) : ext;
```

Comment:
This is an debatable known issue : https://github.com/eslint/eslint/issues/3352
Before it was not getting flagged because function body never took into consideration the variable declaration factor.